### PR TITLE
Update mandrel-packaging to make it work with latest graal

### PR DESCRIPTION
**Pull request by zakkak**
Includes necessary changes to make mandrel-packaging functional after
https://github.com/oracle/graal/commit/e734181ab90e091bc68115591771b5f80a75db5c

### CODE REVIEW

**Comment by zakkak** [comment]
Note to self: Create new branch for `20.1` after merging #111 and before merging this PR.

**Comment by jerboaa** [comment]
> Note to self: Create new branch for `20.1` after merging #111 and before merging this PR.

Yes, please ;-)

File: build.java [hunk 1]
```diff
@@ -263,22 +265,25 @@ private static void patchNativeImageLauncher(Path nativeImage, String mandrelVer
...
                 break;
             }
+            else if(relativeCpMatcher.find())
+            {
+                logger.debugf("Classpath line BEFORE: %s", lines.get(i));
+                logger.debugf("relativeCpMatcher.group(1): %s", relativeCpMatcher.group(1));
+                logger.debugf("relativeCpMatcher.group(2): %s", relativeCpMatcher.group(2));
+                StringBuilder relativeCpLine = new StringBuilder(lines.get(i).length() * 2);
+                relativeCpLine.append(relativeCpMatcher.group(1));
+                relativeCpLine.append(":../../truffle/truffle-api.jar:../../jvmci/graal-sdk.jar");
```
Comment (jerboaa) [inline build.java:282]: Shouldn't this handle the Windows case too?

```
if (IS_WINDOWS) {
...
} else {
     relativeCpLine.append(":../../truffle/truffle-api.jar:../../jvmci/graal-sdk.jar");
}
```

**Review by jerboaa — CHANGES_REQUESTED** [review]

File: build.java (hunk 1 above)
Comment (zakkak) [reply build.java:282]: Correct, I falsely assumed `mx` handles this, but this happens at a prior stage.... Fixing it.

Comment (zakkak) [reply build.java]: Should be fixed in https://github.com/graalvm/mandrel-packaging/pull/112/commits/192bc9327140528be609ab774399baea87f633a5

Comment (jerboaa) [reply build.java]: Thanks!

**Review by jerboaa — CHANGES_REQUESTED** [review]
Smoke tests aren't happy since the version now looks different:

```
./mandrel-build/bin/native-image --version
11.0.11-ea+2 (Java Version 11.0.11-ea+2)
```

**Comment by zakkak** [comment]
That's interesting it looks like `--version` is being passed to the host JVM instead of being passed to `com.oracle.svm.driver.NativeImage`. I am not sure this is mandrel-packaging related. I'll investigate more and get back (probably next week)

**Comment by zakkak** [comment]
Confirmed. https://github.com/oracle/graal/commit/bcf982c95c1955f25c2bdf13b8996fed8a26911d is at fault here. Reverting it prints the correct version...

FTR even with https://github.com/oracle/graal/commit/bcf982c95c1955f25c2bdf13b8996fed8a26911d if we build the native native-image we get the correct version, i.e.:
```
$ /tmp/graal-master/bin/native-image --macro:native-image-launcher
11.0.10+9 (Java Version 11.0.10+9)
$ /tmp/graal-master/bin/native-image --macro:native-image-launcher
[native-image:1504982]    classlist:   1,157.92 ms,  0.96 GB
[native-image:1504982]        (cap):     488.08 ms,  0.96 GB
[native-image:1504982]        setup:   1,906.59 ms,  0.96 GB
[native-image:1504982]     (clinit):     315.85 ms,  2.30 GB
[native-image:1504982]   (typeflow):   7,852.34 ms,  2.30 GB
[native-image:1504982]    (objects):   8,679.96 ms,  2.30 GB
[native-image:1504982]   (features):     637.49 ms,  2.30 GB
[native-image:1504982]     analysis:  17,998.78 ms,  2.30 GB
[native-image:1504982]     universe:     947.54 ms,  2.30 GB
[native-image:1504982]      (parse):   3,177.02 ms,  2.31 GB
[native-image:1504982]     (inline):   2,805.86 ms,  3.16 GB
[native-image:1504982]    (compile):  18,831.36 ms,  3.31 GB
[native-image:1504982]      compile:  25,941.19 ms,  3.31 GB
[native-image:1504982]        image:   2,903.70 ms,  3.31 GB
[native-image:1504982]        write:     312.07 ms,  3.31 GB
# Printing build artifacts to: ../../../../tmp/graal-master/lib/svm/bin/native-image.build_artifacts.txt
[native-image:1504982]      [total]:  51,369.25 ms,  3.31 GB
$ /tmp/graal-master/bin/native-image --version
GraalVM 21.1.0-dev Java 11 CE (Java Version 11.0.10+9)
```

**Comment by jerboaa** [comment]
> Confirmed. [oracle/graal@bcf982c](https://github.com/oracle/graal/commit/bcf982c95c1955f25c2bdf13b8996fed8a26911d) is at fault here. Reverting it prints the correct version...
> 
> FTR even with [oracle/graal@bcf982c](https://github.com/oracle/graal/commit/bcf982c95c1955f25c2bdf13b8996fed8a26911d) if we build the native native-image we get the correct version, i.e.:
> 
> ```
> $ /tmp/graal-master/bin/native-image --macro:native-image-launcher
> 11.0.10+9 (Java Version 11.0.10+9)
> $ /tmp/graal-master/bin/native-image --macro:native-image-launcher
> [native-image:1504982]    classlist:   1,157.92 ms,  0.96 GB
> [native-image:1504982]        (cap):     488.08 ms,  0.96 GB
> [native-image:1504982]        setup:   1,906.59 ms,  0.96 GB
> [native-image:1504982]     (clinit):     315.85 ms,  2.30 GB
> [native-image:1504982]   (typeflow):   7,852.34 ms,  2.30 GB
> [native-image:1504982]    (objects):   8,679.96 ms,  2.30 GB
> [native-image:1504982]   (features):     637.49 ms,  2.30 GB
> [native-image:1504982]     analysis:  17,998.78 ms,  2.30 GB
> [native-image:1504982]     universe:     947.54 ms,  2.30 GB
> [native-image:1504982]      (parse):   3,177.02 ms,  2.31 GB
> [native-image:1504982]     (inline):   2,805.86 ms,  3.16 GB
> [native-image:1504982]    (compile):  18,831.36 ms,  3.31 GB
> [native-image:1504982]      compile:  25,941.19 ms,  3.31 GB
> [native-image:1504982]        image:   2,903.70 ms,  3.31 GB
> [native-image:1504982]        write:     312.07 ms,  3.31 GB
> # Printing build artifacts to: ../../../../tmp/graal-master/lib/svm/bin/native-image.build_artifacts.txt
> [native-image:1504982]      [total]:  51,369.25 ms,  3.31 GB
> $ /tmp/graal-master/bin/native-image --version
> GraalVM 21.1.0-dev Java 11 CE (Java Version 11.0.10+9)
> ```

Sort of. `--version` is being passed to NativeImage, but it then looks up the system property `java.vm.version` which is [replaced]( https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jdk/SystemPropertiesSupport.java#L104) when a native image gets produced for the launcher. That's what [oracle/graal@bcf982c](https://github.com/oracle/graal/commit/bcf982c95c1955f25c2bdf13b8996fed8a26911d)  seems to have changed. The native launcher will print a different version than the bash launcher.

https://github.com/oracle/graal/blob/master/substratevm/src/com.oracle.svm.driver/src/com/oracle/svm/driver/DefaultOptionHandler.java#L84 here:

I'll file an upstream issue asking about the intention of this.

**Comment by jerboaa** [comment]
> I'll file an upstream issue asking about the intention of this.

Asked here:
https://github.com/oracle/graal/issues/3313

**Comment by jerboaa** [comment]
> > I'll file an upstream issue asking about the intention of this.
> 
> Asked here:
> [oracle/graal#3313](https://github.com/oracle/graal/issues/3313)

@zakkak As it's intentional, I'm OK with getting this merged and fixing the smoke test as a follow-up. I've done some preliminary experiments with sdk's `Version` API and it seems to report what `-Dorg.graalvm.version` is being set to. So we should be OK for now for quarkus. Nevertheless, I'd like to have a test which ensures "sane" values are being reported for the `Version` API.

We probably also need to revise the `--version` output or some such. Right now it's not very meaningful with reporting the java version twice :)

**Pull request merged**